Hippo.Web/Controllers/ClusterAdminController.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Linq;
+using Hippo.Core.Data;
+using Hippo.Core.Models;
+using Hippo.Core.Services;
+using Hippo.Web.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace Hippo.Web.Controllers
+{
+    [Authorize(Policy = AccessCodes.SystemAccess)]
+    public class ClusterAdminController : Controller
+    {
+        private readonly ISecretsService _secretsService;
+        private readonly AppDbContext _dbContext;
+
+        public ClusterAdminController(ISecretsService secretsService, AppDbContext dbContext)
+        {
+            _secretsService = secretsService;
+            _dbContext = dbContext;
+        }
+
+        public IActionResult Index()
+        {
+            return View("React");
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Clusters()
+        {
+            var currentSecretKeyIds = new HashSet<string>(await _secretsService.GetSecretNames());
+
+            var clusterModels = (await _dbContext.Clusters
+                .AsNoTracking()
+                .OrderBy(c => c.Name)
+                .ToArrayAsync())
+                .Select(c => new ClusterModel(c, string.Empty))
```

Comment:
this is interesting -- i'm sure there is something that would fail if you didn't, but it seems unfortunate you have to ToArray() which executes the query, and then do a .Select() which could only need a subset of that data.

Reply:
`Where()` would filter to a subset, but not `Select()`